### PR TITLE
Doc fixes around custom types

### PR DESCRIPTION
**Pull request by bendk**
Fixed the CHANGELOG entry for the feature.  It's not released yet, but
it was listed in the changelog as being part of 0.16.0.  Looks like the
wires got crossed because the feature was merged a couple days after
the 0.16.0 release.

Updated the docs to refer to them as "Custom types" instead of "Wrapped
types"

### CODE REVIEW

File: CHANGELOG.md
Comment (jhugman) [inline CHANGELOG.md:9]: Should the `v0.9.0` change here too?

File: CHANGELOG.md
Comment (jhugman) [inline CHANGELOG.md:27]: We've done this in a lot of changes in this file, so I don't really know. Is the `v0.9.0` meant to be previous version?

File: CHANGELOG.md
Comment (jhugman) [inline CHANGELOG.md:23]: Aha, so this hasn't been released yet.

**Review by jhugman — APPROVED** [review]
Two questions about version numbers, but this looks fine. Would another release be in order soon?

File: CHANGELOG.md
Comment (bendk) [reply CHANGELOG.md:27]: That's a really good question.  I think:
  - It should be the previous version
  - `cargo release` should be updating this but it's not.  We should try to fix the automation or at least update the release docs.

Does that makes sense to you?

Comment (bendk) [reply CHANGELOG.md]: I'm really unsure of what happened, but I just pushed a change that fixes the version numbers here.  Maybe once we merge this PR we could make a new release and test if they get updated correctly or not.

Comment (tarikeshaq) [reply CHANGELOG.md]: I think https://github.com/mozilla/uniffi-rs/blob/main/uniffi/release.toml#L11 is the culprit here, it has a constant v0.9.0 😬

Comment (bendk) [reply CHANGELOG.md]: Nice catch.  I'm not sure exactly how that works, but I pushed a blind update to it.  Does it look good to you?

Comment (tarikeshaq) [reply CHANGELOG.md]: Looks right to me (I'm not exactly sure how it works either, but that seems like it should be good 😅) we can land this and cut a release and catch it in action in the release PR

**Comment by bendk** [comment]
Looks like this has approval, I'm going to merge and cross my fingers that the next release has the right version number.

**Pull request merged**
